src/components/IFrame.js

```diff
@@ -97,38 +190,102 @@ const propTypes = {
     }).isRequired,
 };
 
-function OldDotIFrame({session}) {
-    const [oldDotURL, setOldDotURL] = useState('https://staging.expensify.com');
+// Gets the routeName and params of the top most central pane route if it is an iframe route.
+// If the top most central pane route is not an iframe route, then it returns undefined.
+function getIframeRouteNameAndParams(state) {
+    const routeName = getTopMostCentralPaneRouteName(state);
+    if (routeName !== undefined && routeName !== 'Report') {
+        const topMostCentralPane = lodashFindLast(state.routes, (route) => route.name === 'CentralPaneNavigator');
+        if (topMostCentralPane && topMostCentralPane.state) {
+            const params = topMostCentralPane.state.routes[topMostCentralPane.state.index || 0].params;
+            return {routeName, params, index: state.index};
+        }
 
-    useEffect(() => {
-        setOldDotURL(`https://expensify.com.dev/${getOldDotURL(window.location.href)}`);
+        const params = topMostCentralPane.params;
+        if (params) {
+            return {routeName, params: params.params, index: state.index};
+        }
+    }
+    return {routeName: undefined, params: undefined, index: state.index};
+}
+
+// TODO: use proper URL
+const BASE_IFRAME_URL = 'http://localhost:3000';
+
+function IFrame({session}) {
+    const [oldDotURL, setOldDotURL] = useState(undefined);
+    const {routeName, params, index} = useNavigationState((state) => getIframeRouteNameAndParams(state));
 
+    useEffect(() => {
         window.addEventListener('message', (event) => {
-            const url = event.data;
+            let data = {};
+            try {
+                data = JSON.parse(event.data);
+            } catch {
+                data = {};
+            }
+
+            if (data.iframeId !== 'OldDot' || data.type !== 'URL_CHANGED') {
+                return;
+            }
+
+            const {url} = data.payload ?? {};
+
             // TODO: use this value to navigate to a new path
             // eslint-disable-next-line no-unused-vars
             const newDotURL = getNewDotURL(url);
+
+            // TODO: The line below should be enough but I can't test it because of cookies issue.
+            // navigation.navigate(newDotURL);
         });
     }, []);
 
     useEffect(() => {
-        document.cookie = `authToken=${session.authToken}; domain=expensify.com.dev; path=/;`;
-        document.cookie = `email=${session.email}; domain=expensify.com.dev; path=/;`;
+        // The routeName will be undefined if there is no iframe route on the top.
+        // In that case we don't want to change the iframe URL.
+        if (!routeName) {
+            return;
+        }
+
+        const newOldDotURL = getOldDotURL(routeName, params);
+        if (!newOldDotURL) {
+            console.error(`No old dot URL found for routeName: ${routeName} and params: ${JSON.stringify(params)}`);
```

Comment:
We probably don't want to just log this to the console right? We would want to log it to the server and create a bug issue using the `log` class.

Reply:
Also I'm not sure that we want to log to the console at all since that would be user facing